src/entity/Entity.cpp

```diff
@@ -39,16 +39,26 @@ Entity::Entity(nlohmann::json json) {
         texture.loadFromFile(path);
         sprite.setTexture(texture);
     }
-
-
-
 }
 
-Entity::Entity(int ID, std::string name, sf::Vector2f position,sf::RectangleShape hitbox, sf::Sprite sprite) {
+Entity::Entity(int ID, std::string name, sf::Vector2f position, sf::FloatRect hitbox, sf::Texture texture) {
    this->ID = ID;
    this->name = std::move(name);
-   this->sprite = std::move(sprite);
-   this->hitbox = std::move(hitbox);
+   this->texture = texture;
+   this->sprite = sf::Sprite(texture);
+   this->hitbox = hitbox;
    this->position = position;
 }
 
+void Entity::render(sf::RenderWindow *window) {
```

Comment:
I thought we weren't supposed to use plain C++ pointers. Wouldn't a reference or `std::shared_pointer` be better in this case?